sentry-core/src/session.rs

```diff
@@ -184,23 +232,77 @@ impl SessionFlusher {
 
     /// Enqueues a session update for delayed sending.
     ///
-    /// When the queue is full, it will be flushed immediately.
+    /// This will aggregate session counts in request mode, for all sessions
+    /// that were not yet partially sent.
     pub fn enqueue(&self, session_update: SessionUpdate<'static>) {
         let mut queue = self.queue.lock().unwrap();
-        queue.push(session_update);
-        if queue.len() >= MAX_SESSION_ITEMS {
-            SessionFlusher::flush(queue, &self.transport);
+        if self.mode == SessionMode::Application || !session_update.init {
```

Comment:
Shouldn't you always enqueue an update if there is a session? Even in Request mode, one may want to push an individual session with full attributes.